tests/test_execution.py

```diff
@@ -311,7 +260,8 @@ def test_aggregatetuple(global_execution_env):
 
 
 @pytest.mark.slow
-def test_aggregate_composite_traintuples(global_execution_env):
+def test_aggregate_composite_traintuples(factory, clients, default_datasets, default_objectives,
```

Comment:
minor: network should be in first or second position.

Reply:
Yes, it could indeed looks better